cli/credentials.go

```diff
@@ -11,6 +11,31 @@ import (
 	ps "github.com/mitchellh/go-ps"
 )
 
+type ShellType = string
+
+const (
+	shellTypePowershell ShellType = "powershell"
+	shellTypeBash       ShellType = "bash"
+	shellTypeBasic      ShellType = "basic"
+	shellTypeInfer      ShellType = "infer"
+)
+
+func getShellType() ShellType {
+	pid := os.Getppid()
+	parentProc, _ := ps.FindProcess(pid)
+	normalizedName := strings.ToLower(parentProc.Executable())
+
+	if strings.Contains(normalizedName, "powershell") || strings.Contains(normalizedName, "pwsh") {
+		return shellTypePowershell
+	}
+
+	if runtime.GOOS == "windows" {
+		return shellTypeBasic
+	}
+
+	return shellTypeBash
```

Comment:
This is a small change in behaviour for this function, but *not* for the program overall.

Previously, this function would return the normalized name of the process if there was no match, and the caller would override this by just assuming it was Bash

This change just ensures we only ever return Bash.

This is necessary to ensure that we have a fixed set of values being returned from getShellType.

An alternative might be to alter getShellType to return an empty string and a false bool as a second value if the shell is not known/supported